src/patterns/sns-lambda.ts

```diff
@@ -35,8 +36,8 @@ import { GuSnsTopic } from "../constructs/sns";
  *  existingSnsTopic: { externalTopicName: "MySnsTopicNameFromAnotherStack" }
  * ```
  */
-export interface ExistingSnsTopic extends GuMigratingResource {
-  externalTopicName?: string;
+export interface ExistingSnsTopic {
```

Comment:
By dropping `GuMigratingResource` we're pushing the responsibility of persisting a logical ID for the SNS topic to the user. In order to achieve this, we need to make the stream available (see the `public readonly` change below).

Reply:
I think this is following the similar (slightly questionable) API that the `GuKinesisLambda` was using, so we might want to try to improve both simultaneously?

Reply:
I think we can defer this in response to https://github.com/guardian/cdk/pull/1293, so will resolve this thread (i.e. no action needed here, IMO!).